Sources/Orbit/Components/SegmentedSwitch.swift

```diff
@@ -0,0 +1,214 @@
+import SwiftUI
+
+/// Showing two choices from which only one can be selected.
+///
+/// Currently SegmentedSwitch allows to have 2 segments.
+/// SegmentedSwitch can be in error state only in unselected state.
+///
+/// - Note: [Orbit definition](https://orbit.kiwi/components/interaction/segmentedswitch/)
+public struct SegmentedSwitch: View {
+
+    @Binding private var selectedIndex: Int?
+    let label: String
+    let message: Message?
+    let firstOption: String
+    let secondOption: String
+
+    let separatorWidth: CGFloat = BorderWidth.emphasis
+    let borderWidth: CGFloat = BorderWidth.emphasis
+
+    public var body: some View {
+        FieldWrapper(
+            label,
+            message: message
+        ) {
+            HStack(spacing: 0) {
+                segment(title: firstOption)
+                    .accessibility(.segmentedSwitchFirstOption)
+                separator
+                segment(title: secondOption)
+                    .accessibility(.segmentedSwitchSecondOption)
+            }
+            .background(selectionBackground)
+            .background(roundedBackground)
+            .fixedSize(horizontal: false, vertical: true)
+        }
+        .accessibilityElement(children: .ignore)
+        .accessibility(label: .init(label))
+        .accessibility(value: .init(selection?.description ?? ""))
+        .accessibility(hint: .init(accessibilityHint))
+        .accessibility(addTraits: .isButton)
+        .accessibility(addTraits: selection != nil ? .isSelected : [])
+    }
+
+    @ViewBuilder func segment(title: String) -> some View {
+        Text(title)
+            .padding( .xSmall)
+            .frame(maxWidth: .infinity)
+            .contentShape(Rectangle())
+            .onTapGesture {
+                selectedIndex = title == firstOption ? 0 : 1
+            }
+    }
+
+    @ViewBuilder var separator: some View {
+        separatorColor
+          .frame(width: separatorWidth)
+    }
+
+    @ViewBuilder var selectionBackground: some View {
+        HStack {
+            if selection == secondOption {
+                Spacer(minLength: 0)
+                    .layoutPriority(1)
+            }
+
+            Color.whiteDarker
+                .clipShape(RoundedRectangle(cornerRadius: BorderRadius.default - 1))
+                .padding(borderWidth)
+                .elevation(selection != nil ? .level1 : nil)
+                .layoutPriority(1)
+
+            if selection == firstOption {
+                Spacer(minLength: 0)
+                    .layoutPriority(1)
+            }
+        }
+        .animation(.easeOut(duration: 0.2), value: selectedIndex)
+    }
+
+    @ViewBuilder var roundedBackground: some View {
+        RoundedRectangle(cornerRadius: BorderRadius.default)
+            .fill(backgroundColor)
+            .overlay(
```

Comment:
The overlay and `backgroundColor` property should not be necessary - Just the rectangle + fixed `.cloudNormal` color seem to work fine.

Reply:
I've tried to apply overlay to HStack:
```
.background(selectionBackground)
            .background(roundedBackground)
            .overlay(
                RoundedRectangle(cornerRadius: BorderRadius.default)
                    .strokeBorder(borderColor, lineWidth: borderWidth)
            )
            .fixedSize(horizontal: false, vertical: true)
```
However I've got some weird shadow:
![image](https://user-images.githubusercontent.com/5494090/210794499-0b92af76-018b-4f17-881f-009977b012a2.png)

Reply:
This change probably needs to be done together with other suggestions.

Reply:
I think I've applied all of them 🤔 except the separator width....

Reply:
found issue, will be updated